### PR TITLE
Add the release note for 0.99.9

### PR DESCRIPTION
**Pull request by trustin**
- Added the release note for 0.99.0
- Added the `date` to MDX frontmatter and rendered page whenever
  possible.

### CODE REVIEW

**Comment by trustin** [comment]
Preview:
image

File: site/src/pages/release-notes/0.99.9.mdx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  ClientFactory factory =
+      ClientFactory.builder()
+                   .proxyConfig(mySelector)
+                   .build();
+  ```
+- Added <type://Route#patternString()> which can be used as a simple human-readable description of
+  a <type://Route>. #2932 #2933
+- <type://HttpData> now extends `AutoCloseable` and provides various methods useful when using
+  <type://SubscriptionOption#WITH_POOLED_OBJECTS>. #2892
+  - Note that this feature is only for the advanced users who wants to reduce the number of memory copies
```
Comment (minwoox) [inline site/src/pages/release-notes/0.99.9.mdx:120]: nit: wants -> want

File: site/src/pages/release-notes/0.99.9.mdx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+- <type://ClientRequestContext#currentOrNull()> and <type://ServiceRequestContext#currentOrNull()> don't throw
+  an `IllegalStateException` anymore but return `null` when the current context type mismatches. #2962
+- <type://Endpoint#parse(String)> doesn't fail anymore when given with an authority with a userInfo part.
+  #2869 #2893
+- `req.serviceName` in <type://BuiltInProperties> has been renamed to `req.service_name` for the consistency
+  with other property names. #2958
+- We now handle an HTTP/2 request with a `Host` header properly, matching the behavior of HAPROXY. #2877 #2878
+- You can now send an HTTPS request via an HTTPS CONNECT proxy server (double encryption). #2887
+- Fixed the buffer leaks in the following places: 😱 #2891 #2892 #2951
```
Comment (ikhoon) [inline site/src/pages/release-notes/0.99.9.mdx:146]: 😱

File: site/src/pages/release-notes/0.99.9.mdx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  ```java
+  Server.builder()
+        .route().get("/hello")
+                .decorators(firstDecorator,
+                            secondDecorator,
+                            thirdDecorator)
+                .build(helloService)
+        .build();
+  ```
+- You can now specify whether the entire annotated service methods has to run in a `blockingTaskExecutor`.
```
Comment (ikhoon) [inline site/src/pages/release-notes/0.99.9.mdx:22]: has -> have?

File: site/src/pages/release-notes/0.99.9.mdx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+  <type://ServiceRequestContext#requestTimeoutHandler()> and
+  <type://ServiceRequestContext#setRequestTimeoutHandler(Runnable)> have been deprecated in favor of
+  <type://ClientRequestContext#whenResponseTimedOut()> and
+  <type://ServiceRequestContext#whenRequestTimedOut()>. #2939
+
+## ☢️ Breaking changes
+
+- Added `final` to the various classes and methods so that a user doesn't extend them beyond the devs'
+  intention. #2927
+- `com.linecorp.amreria.common.util.NonNullByDefault` and `UnstableApi` have been moved to
```
Comment (KarboniteKream) [inline site/src/pages/release-notes/0.99.9.mdx:183]: `amreria` -> `armeria`

Comment (ikhoon) [reply site/src/pages/release-notes/0.99.9.mdx:183]: 😱

Comment (trustin) [reply site/src/pages/release-notes/0.99.9.mdx:184]: :sweat_smile:

**Comment by ikhoon** [comment]
By the way the PR title is 0.99.0. 😀

**Pull request merged**